qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -287,3 +703,281 @@ def use_high_resolution_adc(self) -> None:
+            n_steps: int,
+            post_sweep_voltage_val: int = constants.WMDCV.Post.STOP,
+            measure_chan_list: Optional[constants.ChannelList] = None,
+            av_coef: int = -1,
+            enable_filter: bool = True,
+            v_src_range: constants.OutputRange = constants.VOutputRange.AUTO,
+            i_comp: float = 10e-6,
+            i_meas_range: Optional[constants.MeasureRange] =
+            constants.IMeasRange.FIX_10uA,
+            hold_delay: float = 0,
+            delay: float = 0,
+            step_delay: float = 0,
+            measure_delay: float = 0,
+            abort_enabled: Union[constants.Abort,
+                                int] = constants.Abort.ENABLED,
+            sweep_mode: Union[constants.SweepMode,
+                                int] = constants.SweepMode.LINEAR
+
+    ):
+        """
+        Setup the staircase sweep measurement using the same set of commands
+        (in the same order) as given in the programming manual - see pages
+        3-19 and 3-20.
+
+        Args:
+            v_start: starting voltage of staircase sweep
+            v_end: ending voltage of staircase sweep
+            n_steps: number of measurement points (uniformly distributed
+                between v_start and v_end)
+            post_sweep_voltage_val: voltage to hold at end of sweep (i.e.
+                start or end val). Sweep chan will also output this voltage
+                if an abort condition is encountered during the sweep
+            measure_chan_list: list of channels to be measured (will be
+                measured in order supplied)
+            av_coef: coefficient to use for av command to set ADC
+                averaging.  Negative value implies NPLC mode with absolute
+                value of av_coeff the NPLC setting to use. Positive value
+                implies auto mode and must be set to >= 4
+            enable_filter: turn SMU filter on or off
+            v_src_range: range setting to use for voltage source
+            i_comp: current compliance level
+            i_meas_range: current measurement range
+            hold_delay: time (in s) to wait before starting very first
+                measurement in sweep
+            delay: time (in s) after starting to force a step output and
+                before starting a step measurement
+            step_delay: time (in s) after starting a step measurement before
+                next step in staircase. If step_delay is < measurement time,
+                B1500 waits until measurement complete and then forces the
+                next step value.
+            measure_delay: time (in s)  after receiving a start step
+                measurement trigger and before starting a step measurement
+            abort_enabled: Enbale abort
+            sweep_mode: Linear, log, linear-2-way or log-2-way
+          """
+        self.set_average_samples_for_high_speed_adc(av_coef)
+        self.average_coefficient = av_coef
```

Comment:
store in a private smth similar to measure/source config

Reply:
Done